api/oc_tcp.c

```diff
@@ -65,3 +74,69 @@ oc_tcp_on_connect_event_free(oc_tcp_on_connect_event_t *event)
 }
 
 #endif /* OC_HAS_FEATURE_TCP_ASYNC_CONNECT */
+
+bool
+oc_tcp_is_valid_header(const oc_message_t *message,
+                       const oc_endpoint_t *endpoint)
+{
+  assert(message != NULL);
+  assert(endpoint != NULL);
+#ifdef OC_SECURITY
+  if ((endpoint->flags & SECURED) != 0) {
+    if (message->length < 3) {
+      OC_ERR("TLS header too short: %lu", (long unsigned)message->length);
+      // TLS header too short
+      return false;
+    }
+    // Parse the header fields
+    uint8_t type = message->data[0];
+    uint8_t major_version = message->data[1];
+    uint8_t minor_version = message->data[2];
+    OC_DBG("TLS header: record type: %d, major %d, minor %d", type,
+           major_version, minor_version);
+    // Validate the header fields
+    switch (type) {
+    case MBEDTLS_SSL_MSG_HANDSHAKE:
+    case MBEDTLS_SSL_MSG_CHANGE_CIPHER_SPEC:
+    case MBEDTLS_SSL_MSG_ALERT:
+    case MBEDTLS_SSL_MSG_APPLICATION_DATA:
+    case MBEDTLS_SSL_MSG_CID:
+      // Valid record type
+      break;
+    default:
+      OC_ERR("invalid record type: %d", type);
+      // Invalid record type
+      return false;
+    }
+    if (major_version != MBEDTLS_SSL_MAJOR_VERSION_3) {
+      OC_ERR("invalid major version: %d", major_version);
+      // Invalid major version
+      return false;
+    }
+    if (minor_version != MBEDTLS_SSL_MINOR_VERSION_3 &&
+        minor_version != MBEDTLS_SSL_MINOR_VERSION_4 &&
+        // TLS 1.0
+        minor_version != 1 &&
+        // TLS 1.1
+        minor_version != 2) {
+      OC_ERR("invalid minor version: %d", minor_version);
+      // TLS 1.2 and TLS 1.3 are supported
```

Comment:
TLS 1.0 and TLS 2.0 are supported too if the comments on lines 118 and 120 are correct